internal/memdb/memdb_test.go

```diff
@@ -3,199 +3,101 @@ package memdb
-
-	if err == nil {
-		t.Fatal("expected error")
-	}
-}
-
-func TestOnWriteError(t *testing.T) {
-	ctx := context.Background()
-	newClientFunc := GetNewClient(WithErrOnWrite())
-	p := destination.NewPlugin("test", "development", newClientFunc)
-	if err := p.Init(ctx, getTestLogger(t), specs.Destination{}); err != nil {
-		t.Fatal(err)
-	}
-	table := schema.TestTable("test", schema.TestSourceOptions{})
-	tables := schema.Tables{
-		table,
-	}
-	sourceName := "TestDestinationOnWriteError"
-	syncTime := time.Now()
-	sourceSpec := specs.Source{
-		Name: sourceName,
-	}
-	ch := make(chan arrow.Record, 1)
-	opts := schema.GenTestDataOptions{
-		SourceName: "test",
-		SyncTime:   time.Now(),
-		MaxRows:    1,
-		StableUUID: uuid.Nil,
-	}
-	record := schema.GenTestData(table, opts)[0]
-	ch <- record
-	close(ch)
-	err := p.Write(ctx, sourceSpec, tables, syncTime, ch)
-	if err == nil {
-		t.Fatal("expected error")
-	}
-	if err.Error() != "errOnWrite" {
-		t.Fatalf("expected errOnWrite, got %s", err.Error())
-	}
-}
-
-func TestOnWriteCtxCancelled(t *testing.T) {
-	ctx := context.Background()
-	newClientFunc := GetNewClient(WithBlockingWrite())
-	p := destination.NewPlugin("test", "development", newClientFunc)
-	if err := p.Init(ctx, getTestLogger(t), specs.Destination{}); err != nil {
-		t.Fatal(err)
-	}
-	table := schema.TestTable("test", schema.TestSourceOptions{})
-	tables := schema.Tables{
-		table,
-	}
-	sourceName := "TestDestinationOnWriteError"
-	syncTime := time.Now()
-	sourceSpec := specs.Source{
-		Name: sourceName,
-	}
-	ch := make(chan arrow.Record, 1)
-	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
-	opts := schema.GenTestDataOptions{
-		SourceName: "test",
-		SyncTime:   time.Now(),
-		MaxRows:    1,
-		StableUUID: uuid.Nil,
-	}
-	record := schema.GenTestData(table, opts)[0]
-	ch <- record
-	defer cancel()
-	err := p.Write(ctx, sourceSpec, tables, syncTime, ch)
-	if err != nil {
+	p := plugin.NewPlugin("test", "development", NewMemDBClient)
+	if err := p.Init(ctx, nil); err != nil {
 		t.Fatal(err)
 	}
-}
-
-func TestPluginInit(t *testing.T) {
-	const (
-		batchSize      = 100
-		batchSizeBytes = 1000
-	)
-
-	var (
-		batchSizeObserved      int
-		batchSizeBytesObserved int
-	)
-	p := destination.NewPlugin(
-		"test",
-		"development",
-		func(ctx context.Context, logger zerolog.Logger, s specs.Destination) (destination.Client, error) {
-			batchSizeObserved = s.BatchSize
-			batchSizeBytesObserved = s.BatchSizeBytes
-			return NewClient(ctx, logger, s)
+	plugin.TestWriterSuiteRunner(
+		t,
+		p,
+		plugin.PluginTestSuiteTests{
+			SafeMigrations: plugin.SafeMigrations{},
 		},
-		destination.WithDefaultBatchSize(batchSize),
-		destination.WithDefaultBatchSizeBytes(batchSizeBytes),
 	)
-	require.NoError(t, p.Init(context.TODO(), getTestLogger(t), specs.Destination{}))
-
-	require.Equal(t, batchSize, batchSizeObserved)
-	require.Equal(t, batchSizeBytes, batchSizeBytesObserved)
 }
+
+// func TestPluginOnNewError(t *testing.T) {
```

Comment:
Can you explain the reason for commenting these tests? Maybe remove them if not needed